app/assets/src/components/views/bulk_download/BulkDownloadModal.jsx

```diff
@@ -58,38 +28,44 @@ class BulkDownloadModal extends React.Component {
     // in the params. This is also how the bulk download is stored in the database.
     selectedFieldsDisplay: {},
     selectedDownloadTypeName: null,
-    currentStep: "choose",
     validSampleIds: new Set(),
     invalidSampleNames: [],
     validationError: null,
+    backgroundOptions: null,
+    metricsOptions: null,
+    allSamplesUploadedByCurrentUser: false,
+    loading: true,
   };
 
   componentDidMount() {
-    this.fetchDownloadTypesAndValidationInfo();
+    this.parallelFetchNeededData();
   }
 
-  componentDidUpdate(prevProps) {
-    // If the user has just closed the modal, reset it.
-    if (prevProps.open && !this.props.open) {
-      this.setState({
-        selectedDownloadTypeName: null,
-        currentStep: "choose",
-        selectedFields: {},
-      });
-    }
-  }
+  // *** Async requests ***
 
-  async fetchDownloadTypesAndValidationInfo() {
+  async parallelFetchNeededData() {
```

Comment:
Can we have a more explicity name?
Using `parallel` is less important than trying to say what are we fetching.